src/engine.ml

```diff
@@ -366,6 +367,13 @@ let maybe_kex_client rng config tls =
       in
       Some [ "IV_PLAT=mirage"; "IV_CIPHERS=" ^ ciphers; "IV_NCP=2" ]
     in
+    let peer_info =
+      match peer_info with
+      | Some peer_info when pull ->
+          Some ("IV_PROTO=4" (* IV_PROTO_REQUEST_PUSH *) :: peer_info)
```

Comment:
At some point we probably want to generate this from a bitmap, but for now it's fine.